src/timessquare/storage/nbhtmlcache.py

```diff
@@ -0,0 +1,139 @@
+"""The Redis-based cache of rendered HTML pages."""
+
+from __future__ import annotations
+
+from typing import Any, AsyncIterable, Mapping, Optional
+
+import aioredis
+
+from timessquare.domain.nbhtml import NbHtmlModel
+
+from .redisutils import calculate_key
+
+
+class NbHtmlCacheStore:
+    """Manages the storage of HTML renderings of notebooks.
+
+    The domain is `timessquare.domain.nbhtml.NbHtmlModel`.
+    """
+
+    def __init__(self, redis: aioredis.Redis) -> None:
+        self._redis = redis
+
+    async def store(
+        self, nbhtml: NbHtmlModel, lifetime: Optional[int] = None
+    ) -> None:
+        """Store an HTML page.
+
+        Parameters
+        ----------
+        nbhtml : `timessquare.domain.nbhtml.NbHtmlModel`
+            The HTML page domain model.
+        lifetime : int, optional
+            The lifetime for the record. `None` to cache the record
+            indefinitely.
+        """
+        key = NbHtmlCacheStore.calculate_key(
```

Comment:
It looks weird to me to call a static method on one's own class using the full class name, and in the unlikely event that you would ever want to create a derivative class, it will also do the wrong thing with inheritance. I think this here and elsewhere (and also in the other job store) should just be `self.calculate_key`.